### PR TITLE
Added minecraft-launcher profile

### PR DESCRIPTION
**Pull request by kortewegdevries**
Request:# 1139

<s>What to do with `private-etc`? In game multiplayer or realms doesn't work with it, tried with many combinations. Single player works fine with/without.</s> Should be okay now.

### CODE REVIEW

File: etc/profile-m-z/minecraft-launcher.profile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# Persistent local customizations
+include minecraft-launcher.local
+# Persistent global definitions
+include globals.local
+
+ignore noexec ${HOME}
+
+noblacklist ${HOME}/.minecraft
+
+include allow-java.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/minecraft-launcher.profile:13]: Own block; add a blank line.

File: etc/profile-m-z/minecraft-launcher.profile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+netfilter
+nogroups
+nonewprivs
+noroot
+notv
+nou2f
+novideo
+protocol unix,inet,inet6,netlink
+seccomp
+shell none
```
Comment (rusty-snake) [inline etc/profile-m-z/minecraft-launcher.profile:41]: Can we add `tracelog`?

File: etc/profile-m-z/minecraft-launcher.profile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+whitelist ${HOME}/.minecraft
+include whitelist-common.inc
+include whitelist-runuser-common.inc
+include whitelist-usr-share-common.inc
+include whitelist-var-common.inc
+
+apparmor
+caps.drop all
+netfilter
+nogroups
```
Comment (rusty-snake) [inline etc/profile-m-z/minecraft-launcher.profile:33]: `nodvd`?

File: etc/profile-m-z/minecraft-launcher.profile (hunk 1 above)
Comment (kortewegdevries) [reply etc/profile-m-z/minecraft-launcher.profile:13]: Below allow-java?

File: etc/profile-m-z/minecraft-launcher.profile (hunk 2 above)
Comment (kortewegdevries) [reply etc/profile-m-z/minecraft-launcher.profile:41]: The launcher starts but not the game. I'm going to try again,if it works I'll add it.

File: etc/profile-m-z/minecraft-launcher.profile (hunk 1 above)
Comment (rusty-snake) [reply etc/profile-m-z/minecraft-launcher.profile:13]: yes

File: etc/profile-m-z/minecraft-launcher.profile (hunk 2 above)
Comment (kortewegdevries) [reply etc/profile-m-z/minecraft-launcher.profile:43]: I misremembered I think.

File: etc/profile-m-z/minecraft-launcher.profile (hunk 3 above)
Comment (kortewegdevries) [reply etc/profile-m-z/minecraft-launcher.profile:35]: Yeah missed it,added.

File: etc/profile-m-z/minecraft-launcher.profile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+protocol unix,inet,inet6,netlink
+seccomp
+shell none
+tracelog
+
+disable-mnt
+private-bin minecraft-launcher,java,java-config
+private-cache
+private-dev
+# private-etc alternatives,asound.conf,ca-certificates,crypto-policies,fonts,group,gtk-2.0,gtk-3.0,host.conf,hostname,hosts,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,localtime,machine-id,mime.types,nvidia,passwd,pki,pulse,resolv.conf,services,selinux,ssl,xdg,X11
```
Comment (Fred-Barclay) [inline etc/profile-m-z/minecraft-launcher.profile:50]: What happens if this line is uncommented?

Comment (kortewegdevries) [reply etc/profile-m-z/minecraft-launcher.profile:50]: Can't connect to a server or Realms.

**Comment by kortewegdevries** [comment]
I should whitelist `/opt/minecraft-launcher` since swiftshader cache is there.

**Comment by rusty-snake** [comment]
Why? You has already `private-opt minecraft-launcher`.

**Comment by kortewegdevries** [comment]
Adding a `private-opt` copies the entire subdirectory?

**Comment by rusty-snake** [comment]
Ahh you mean removing private-opt and using whitelist instead.

**Comment by kortewegdevries** [comment]
The launcher has this bug where it doesn't create the swiftshader folder in `/opt/...`, I fixed this in my main system (with this firejail profile) following https://bugs.mojang.com/browse/MCL-12546?focusedCommentId=642275&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-642275 but I can't get the same thing to work under a different testing machine. Tried both ways, either it is because of the launcher itself or I'm not whitelisting the directory or something about the libraries.

**Comment by rusty-snake** [comment]
anything left?

**Comment by kortewegdevries** [comment]
> anything left?

Deciding on `private-etc` ,I guess. Are multiplayer known to cause problems with it: minetest,steam  etc.?

**Comment by rusty-snake** [comment]
How do you created it?

**Comment by kortewegdevries** [comment]
I allowed the whole package of files from the template, still didn't work. I made it using the default `private-etc` that goes with an application with sound,video,internet and depending on terminal spam added the rest like `passwd`. Maybe I'll try with strace.

**Comment by rusty-snake** [comment]
First try --build, much easier. The templates don't cover special things. In general: always check if there is a /etc/<minecraft>.

**Comment by kortewegdevries** [comment]
Build gave `private-etc resolv.conf,login.defs,openal,java-13-openjdk,drirc,timezone,fonts,xdg,gtk-3.0,machine-id,ssl,ati,hosts,selinux,passwd`

**Comment by rusty-snake** [comment]
I tend to `java-13-openjdk`, if so add it for more java versions.

**Comment by kortewegdevries** [comment]
> I tend to java-13-openjdk, if so add it for more java versions.

No I checked again removing it, works.

**Comment by kortewegdevries** [comment]
> I tend to java-13-openjdk, if so add it for more java versions.

Nevermind, it was java, I removed the wrong group. How do I add more java versions? Like `java-8-...`,`java-11-...` etc. ?

**Comment by rusty-snake** [comment]
Look at i2prouter, signal-cli, terasology. There is no realy good way now.

**Comment by kortewegdevries** [comment]
I'll add all the known ones of OpneJDK from 7-14. So a comment saying to comment this line in case breaks or add your own java folder?

**Comment by rusty-snake** [comment]
The profile needs sorting (private-bin, private-etc).

**Comment by kortewegdevries** [comment]
Used the python script on it.

**Comment by rusty-snake** [comment]
+1 from me.

File: etc/profile-m-z/minecraft-launcher.profile (hunk 4 above)
Comment (kortewegdevries) [reply etc/profile-m-z/minecraft-launcher.profile]: This should be resolved now can you please check,thanks,

Comment (kortewegdevries) [reply etc/profile-m-z/minecraft-launcher.profile]: > Can't connect to a server or Realms.

This should be resolved now can you please check,thanks,

**Comment by rusty-snake** [comment]
Merged, Thanks.

**Pull request merged**